packages/web-media-box/src/dash-parser/nodes/nodesWithAttributes.ts

```diff
@@ -0,0 +1,50 @@
+import type { ParsedManifest } from '../types/parsedManifest';
+import { NodeProcessor } from './nodeProcessor';
+import { missingRequiredAttributeWarn } from '../utils/warn';
+import { parseAttributes } from '../parseAttributes';
+
+export abstract class NodeWithAttributesProcessor extends NodeProcessor {
+  public process(manifest: ParsedManifest): void {
+
+    // create attributes from expected attributes
+    const atts: Record<string, unknown> = {};
+
+    this.expectedAttributes.forEach((attribute) => {
+      const value = this.node.getAttribute(attribute.name);
+
+      if (attribute.required && !value) {
```

Comment:
Is it possible that an attribute does not have value, for example, boolean attributes?
Is it better to check explicit null?